src/Hl7.Fhir.Core/Model/ModelInfo.cs

```diff
@@ -412,6 +412,39 @@ public static bool IsProfiledQuantity(string type)
             return IsProfiledQuantity(definedType.Value);
         }
 
+        public static bool CheckMinorVersionCompatibility(string externalVersion)
+        {
+            if (string.IsNullOrEmpty(externalVersion))
+            {
+                return false;
+            }
+
+            var minorFhirVersion = GetMinorVersion(Version);
+            var externalMinorVersion = GetMinorVersion(externalVersion);
+
+            if (string.IsNullOrEmpty(minorFhirVersion) || string.IsNullOrEmpty(externalVersion))
+            {
+                return false;
+            }
+            else
+            {
+                return minorFhirVersion == externalMinorVersion;
+            }
+        }
+
+        public static string GetMinorVersion(string version)
+        {
+            var versionnumbers = version.Split('.');
```

Comment:
This crashes when you pass it a null =>  better make this `private` anyway, the public surface of ModelInfo is big enough already.